src/main/java/com/fasterxml/jackson/databind/ser/std/NumberSerializer.java

```diff
@@ -69,6 +102,10 @@ public JsonSerializer<?> createContextual(SerializerProvider prov,
     @Override
     public void serialize(Number value, JsonGenerator g, SerializerProvider provider) throws IOException
     {
+        if (_format != null) {
```

Comment:
```suggestion
        // [databind#1114] since 2.17 : Support @JsonFormat for String, numbers, using String.format()
        if (_format != null) {
```